src/tiled.lisp

```diff
@@ -253,27 +252,29 @@
             (lambda (l)
               (etypecase l
                 (ttile-layer (%load-tile-layer l ret nil))
-                (tobject-group (%load-object-layer l ret nil))
+                (tobject-group (%load-object-layer l ret nil resource-loader))
                 (timage-layer (%load-image-layer l ret nil))
-                (tlayer-group (%load-layer-group l ret nil))))
+                (tlayer-group (%load-layer-group l ret nil resource-loader))))
             (tmap-layers tmap))))
     (setf (slot-value ret 'layers) loaded-layers)
     ret))
 
+
 (defun load-map (path &optional (resource-loader #'read-file-into-string)
-                 &aux
-                   (tmap
-                    (eswitch ((pathname-type path) :test 'string-equal)
-                      ("tmx"
-                       (with-input-from-string (stream (funcall resource-loader path))
-                         (parse-xml-map-stream stream path)))
-                      ("json"
-                       (with-input-from-string (stream (funcall resource-loader path))
-                         (parse-json-map-stream stream path))))))
-  (%load-map tmap path resource-loader))
-
-(defun load-tileset (path)
-  (%load-external-tileset path 0 #'read-file-into-string))
+                 &aux (string (funcall resource-loader path)))
```

Comment:
Relatively minor, but could you remove the aux var, and do
```lisp
(with-input-from-string (stream (funcall resource-loader path))
  ...)
  ```
  in each case at the bottom? The `funcall` code will be duplicated, but the advantage is that we check (via the `eswitch`)` the `pathname-type` of path before we try and load the data from it.
This can be good sanity checking in case some other file path makes it in there, like a zip

Reply:
Ah right, that checking slipped my mind. The only problem is that both `resource-loader` should be called with initial `*default-pathnames-default*`, otherwise in the case of relative map paths it would try something awkward, like reading file `"path/to/map/path/to/map/map.tmx"` (with path duplicated). I guess I'll duplicate `with-pathname-defaults` and `%load-map` too, not reading the file when there's an error is important enough.